contracts/serverless-v2/Relay.sol

```diff
@@ -499,11 +630,620 @@ contract Relay is
+            revert RelayNotJobSubscriptionOwner();
+
+        if(block.timestamp <= jobSubscriptions[_jobSubsId].terminationTimestamp + OVERALL_TIMEOUT)
+            revert RelayTerminationConditionPending();
+
+        uint256 usdcAmount = jobSubscriptions[_jobSubsId].job.usdcDeposit;
+        uint256 callbackAmount = jobSubscriptions[_jobSubsId].job.callbackDeposit;
+
+        delete jobSubscriptions[_jobSubsId];
+
+        TOKEN.safeTransfer(_jobOwner, usdcAmount);
+        // TODO: do we need to check this bool success
+        (bool success, ) = _jobOwner.call{value: callbackAmount}("");
+
+        emit JobSubscriptionFundsWithdrawn(_jobSubsId, _jobOwner, usdcAmount, callbackAmount, success);
+    }
+
+    function _terminateJobSubscription(
+        uint256 _jobSubsId,
+        address _jobOwner
+    ) internal {
+        if(jobSubscriptions[_jobSubsId].job.jobOwner != _jobOwner)
+            revert RelayNotJobSubscriptionOwner();
+
+        if(block.timestamp > jobSubscriptions[_jobSubsId].terminationTimestamp)
+            revert RelayJobSubscriptionTerminated();
+
+        jobSubscriptions[_jobSubsId].terminationTimestamp = block.timestamp;
+
+        emit JobSubscriptionTerminated(_jobSubsId);
+    }
+
+    //-------------------------------- internal functions end --------------------------------//
+
+    //-------------------------------- external functions start --------------------------------//
+
+    /**
+     * @notice Starts a subscription for periodic job execution.
+     * @dev The subscription parameters are validated, and the necessary deposits(USDC+ETH) are made.
+     * @param _codehash The transaction hash storing the code to be executed periodically.
+     * @param _codeInputs The inputs to the code to be executed periodically.
+     * @param _userTimeout The maximum execution time allowed for each job in milliseconds.
+     * @param _maxGasPrice The maximum gas price the subscriber is willing to pay to get back the job response.
+     * @param _refundAccount The account to receive any remaining/slashed tokens.
+     * @param _callbackContract The contract address to be called upon submitting job response.
+     * @param _callbackGasLimit The gas limit for the callback function.
+     * @param _periodicGap The time gap between each job relay in milliseconds.
+     * @param _usdcDeposit The amount of USDC to be deposited for the subscription.
+     * @param _startTimestamp The timestamp at which the job subscription would activate and start relaying periodic jobs.
+     * @param _terminationTimestamp The timestamp after which no further jobs are relayed.
+     */
+    function startJobSubscription(
+        bytes32 _codehash,
+        bytes calldata _codeInputs,
+        uint256 _userTimeout,
+        uint256 _maxGasPrice,
+        address _refundAccount,
+        address _callbackContract,
+        uint256 _callbackGasLimit,
+        uint256 _periodicGap,
+        uint256 _usdcDeposit,
+        uint256 _startTimestamp,
```

Comment:
We should add a way for user to specify the immediate startTimestamp. one way could be if user provide 0 as startTimestamp, then treat that as block.timestamp